### PR TITLE
Feature/add network and common modules

### PR DESCRIPTION
**Pull request by jakobuid**
Added Network and Common modules to `android_core`.

### CODE REVIEW

**Review by TalhaAli00 — COMMENTED** [review]
@jakobuid can you tag the issue this PR addresses?

File: signSDK/sign/src/test/kotlin/com/walletconnect/sign/adapters/ExpiryAdapterTest.kt [hunk 1]
```diff
@@ -1,9 +1,9 @@
-package com.walletconnect.sign.common.network.adapters
+package com.walletconnect.sign.adapters
```
Comment (Elyniss) [inline signSDK/sign/src/test/kotlin/com/walletconnect/sign/adapters/ExpiryAdapterTest.kt:1]: I think this test should be in `android_core` module

File: signSDK/sign/src/test/kotlin/com/walletconnect/sign/adapters/MappingFunctionsTest.kt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package com.walletconnect.sign.common
+package com.walletconnect.sign.adapters
 
 internal class MappingFunctionsTest {
```
Comment (Elyniss) [inline signSDK/sign/src/test/kotlin/com/walletconnect/sign/adapters/MappingFunctionsTest.kt:3]: Do we need this class?

File: signSDK/sign/src/test/kotlin/com/walletconnect/sign/core/adapters/RelayerDOJsonRpcResponseJsonRpcResultJsonAdapterTest.kt [hunk 3]
```diff
@@ -2,10 +2,11 @@ package com.walletconnect.sign.core.adapters
 
 import com.squareup.moshi.Moshi
 import com.tinder.scarlet.utils.getRawType
+import com.walletconnect.android_core.common.adapters.RelayDOJsonRpcResultJsonAdapter
```
Comment (Elyniss) [inline signSDK/sign/src/test/kotlin/com/walletconnect/sign/core/adapters/RelayerDOJsonRpcResponseJsonRpcResultJsonAdapterTest.kt:5]: I think this test should be in `android_core` module

File: signSDK/sign/src/test/kotlin/com/walletconnect/sign/crypto/codec/ChaChaPolyCodecTest.kt [hunk 4]
```diff
@@ -1,10 +1,10 @@
 package com.walletconnect.sign.crypto.codec
 
 import com.walletconnect.sign.core.exceptions.client.WalletConnectException
-import com.walletconnect.sign.core.model.type.enums.EnvelopeType
+import com.walletconnect.android_core.common.model.type.enums.EnvelopeType
 import com.walletconnect.sign.core.model.vo.PublicKey
 import com.walletconnect.sign.core.model.vo.SymmetricKey
-import com.walletconnect.sign.core.model.vo.TopicVO
+import com.walletconnect.sign.core.model.vo.Topic
```
Comment (Elyniss) [inline signSDK/sign/src/test/kotlin/com/walletconnect/sign/crypto/codec/ChaChaPolyCodecTest.kt:7]: wrong import. Should be `import com.walletconnect.foundation.common.model.Topic`

File: signSDK/sign/src/test/kotlin/com/walletconnect/sign/crypto/data/repository/BouncyCastleCryptoRepositoryTest.kt [hunk 5]
```diff
@@ -2,7 +2,7 @@ package com.walletconnect.sign.crypto.data.repository
 
 import com.walletconnect.sign.core.model.vo.PrivateKey
 import com.walletconnect.sign.core.model.vo.PublicKey
-import com.walletconnect.sign.core.model.vo.TopicVO
+import com.walletconnect.sign.core.model.vo.Topic
```
Comment (Elyniss) [inline signSDK/sign/src/test/kotlin/com/walletconnect/sign/crypto/data/repository/BouncyCastleCryptoRepositoryTest.kt:5]: Wring import. Should be `import com.walletconnect.foundation.common.model.Topic`

File: signSDK/sign/src/test/kotlin/com/walletconnect/sign/json_rpc/domain/RelayerInteractorTest.kt [hunk 6]
```diff
@@ -3,18 +3,18 @@ package com.walletconnect.sign.json_rpc.domain
 import com.walletconnect.sign.core.exceptions.client.WalletConnectException
 import com.walletconnect.sign.core.exceptions.peer.PeerError
 import com.walletconnect.sign.core.model.client.Relay
```
Comment (Elyniss) [inline signSDK/sign/src/test/kotlin/com/walletconnect/sign/json_rpc/domain/RelayerInteractorTest.kt:5]: wrong import

File: signSDK/sign/src/test/kotlin/com/walletconnect/sign/json_rpc/domain/RelayerInteractorTest.kt [hunk 7]
```diff
@@ -3,18 +3,18 @@ package com.walletconnect.sign.json_rpc.domain
...
-import com.walletconnect.sign.core.model.type.ClientParams
-import com.walletconnect.sign.core.model.type.JsonRpcClientSync
-import com.walletconnect.sign.core.model.type.enums.Tags
+import com.walletconnect.android_core.common.model.type.ClientParams
+import com.walletconnect.android_core.common.model.type.JsonRpcClientSync
+import com.walletconnect.android_core.common.model.type.enums.Tags
 import com.walletconnect.sign.core.model.vo.IrnParamsVO
-import com.walletconnect.sign.core.model.vo.TopicVO
+import com.walletconnect.sign.core.model.vo.Topic
```
Comment (Elyniss) [inline signSDK/sign/src/test/kotlin/com/walletconnect/sign/json_rpc/domain/RelayerInteractorTest.kt:10]: wrong import

File: signSDK/sign/src/test/kotlin/com/walletconnect/sign/json_rpc/domain/RelayerInteractorTest.kt [hunk 8]
```diff
@@ -3,18 +3,18 @@ package com.walletconnect.sign.json_rpc.domain
...
 import com.walletconnect.sign.core.model.vo.TtlVO
 import com.walletconnect.sign.core.model.vo.jsonRpc.JsonRpcResponseVO
 import com.walletconnect.sign.core.model.vo.sync.WCRequestVO
 import com.walletconnect.sign.crypto.data.codec.ChaChaPolyCodec
 import com.walletconnect.sign.json_rpc.data.JsonRpcSerializer
-import com.walletconnect.sign.network.RelayInterface
-import com.walletconnect.sign.network.model.RelayDTO
+import com.walletconnect.android_core.network.RelayConnectionInterface
+import com.walletconnect.android_core.network.model.RelayDTO
```
Comment (Elyniss) [inline signSDK/sign/src/test/kotlin/com/walletconnect/sign/json_rpc/domain/RelayerInteractorTest.kt:17]: wrong import

File: signSDK/sign/src/test/kotlin/com/walletconnect/sign/ValidatorTest.kt [hunk 9]
```diff
@@ -25,7 +25,7 @@ import com.walletconnect.sign.ValidatorTest.Namespaces.EIP155
 import com.walletconnect.sign.core.exceptions.*
 import com.walletconnect.sign.core.model.utils.Time
 import com.walletconnect.sign.core.model.vo.SymmetricKey
-import com.walletconnect.sign.core.model.vo.TopicVO
+import com.walletconnect.sign.core.model.vo.Topic
```
Comment (Elyniss) [inline signSDK/sign/src/test/kotlin/com/walletconnect/sign/ValidatorTest.kt:28]: wrong import

**Review by Elyniss — CHANGES_REQUESTED** [review]

File: signSDK/sign/src/test/kotlin/com/walletconnect/sign/adapters/MappingFunctionsTest.kt (hunk 2 above)
Comment (jakobuid) [reply signSDK/sign/src/test/kotlin/com/walletconnect/sign/adapters/MappingFunctionsTest.kt:3]: No, removed

File: signSDK/sign/src/main/kotlin/com/walletconnect/sign/core/adapters/SubscriptionIdAdapter.kt [hunk 10]
```diff
@@ -1,39 +0,0 @@
-package com.walletconnect.sign.core.adapters
-
-import com.squareup.moshi.*
-import com.walletconnect.sign.core.model.vo.SubscriptionIdVO
-
-internal object SubscriptionIdAdapter: JsonAdapter<SubscriptionIdVO>() {
```
[the remark this answers is not in the document]
Comment (jakobuid) [reply signSDK/sign/src/main/kotlin/com/walletconnect/sign/core/adapters/SubscriptionIdAdapter.kt:6]: This adapter was moved to `foundation`. There is no point to keep this class in `sign`

File: signSDK/sign/src/test/kotlin/com/walletconnect/sign/adapters/SubscriptionIdAdapterTest.kt [hunk 11]
```diff
@@ -1,4 +1,4 @@
-package com.walletconnect.sign.common.network.adapters
+package com.walletconnect.sign.adapters
```
[the remark this answers is not in the document]
Comment (jakobuid) [reply signSDK/sign/src/test/kotlin/com/walletconnect/sign/adapters/SubscriptionIdAdapterTest.kt:1]: Yes, good call. I moved it to `foundation`

File: signSDK/sign/src/main/kotlin/com/walletconnect/sign/core/adapters/TopicAdapter.kt [hunk 12]
```diff
@@ -1,39 +0,0 @@
-package com.walletconnect.sign.core.adapters
-
-import com.squareup.moshi.*
-import com.walletconnect.sign.core.model.vo.TopicVO
-
-internal object TopicAdapter: JsonAdapter<TopicVO>() {
```
[the remark this answers is not in the document]
Comment (jakobuid) [reply signSDK/sign/src/main/kotlin/com/walletconnect/sign/core/adapters/TopicAdapter.kt:6]: This adapter was moved to foundation. There is no point to keep this class in sign

File: signSDK/sign/src/test/kotlin/com/walletconnect/sign/json_rpc/domain/RelayerInteractorTest.kt (hunk 8 above)
Comment (jakobuid) [reply signSDK/sign/src/test/kotlin/com/walletconnect/sign/json_rpc/domain/RelayerInteractorTest.kt:17]: This import is fine. Once RelayerInteractor is moved to `android_core` it'll compile

**Review by Elyniss — APPROVED** [review]

**Review by TalhaAli00 — APPROVED** [review]

**Pull request merged**